### ISSUE TITLE
Update to MongoDB C++ Driver 3.0.0

### ISSUE THREAD

**Issue by jozefdobos** [3drepo/3drepobouncer#83]
https://github.com/mongodb/mongo-cxx-driver/releases

Once done, make sure to update also our own instructions. These at the moment point to 1.0.5 legacy: https://github.com/3drepo/3drepobouncer/wiki/Compiling-MongoDB-CXX-Driver

**Comment by carmenfan** [comment]
NOTE: It looks like the c++11 driver uses CMake instead of scons so be sure to update travis script so it compiles for autotests

**Comment by jozefdobos** [comment]
Ok, that would explain why scons was complaining so badly last night!

**Comment by carmenfan** [comment]
In short, the entire API and the way it works is completely different to legacy CXX driver.

This issue is going to be parked until mongodb guys release a more developed version of the driver for 2 main reasons:

1. the API isn't set in stone yet, they want to get feedback from users and see how they should go about modifying it, if at all
2. It currently feels really green, there is hardly any documentation on the driver apart from the API. Only other thing out there is a brief tutorial which has very brief example of how to do simple things. 2 biggest concerns I've encountered so far is 

1.  the only way to connect and authenticate a user is to pass in a connection string in plain text, i.e. there is no way for us to store the digested password, and there is no way for us to keep a connection and change authentication (there seems to be some todos within the code on auth part within the database objection)
2. the connection procedure has no way of testing if you have been successfully authenticated or connected on the database until you try to do an operation that causes an exception to be thrown. For example I put in the right database address but wrong authentication, it "connects" fine, can list the databases (non existent) and not until i try to insert a document does it throw an exception. 

Granted everything can be worked around, the runCommand() function is available on the API so we could fill in the gaps ourselves. But it is probbaly not the best way to go as it is almost as if we're writing our own interface to the database if we rely on runCommand() and will be doing extra work when they decide to change their API
